pkg/reconciler/trigger/trigger.go

```diff
@@ -236,6 +252,71 @@ func (r *Reconciler) reconcileRetryTopicAndSubscription(ctx context.Context, tri
 	return nil
 }
 
+// getPubsubRetryPolicy gets the eventing retry policy from the Broker delivery
+// spec and translates it to a pubsub retry policy.
+func getPubsubRetryPolicy(spec *eventingduckv1beta1.DeliverySpec) (*pubsub.RetryPolicy, error) {
+	if spec == nil || spec.BackoffDelay == nil || spec.BackoffPolicy == nil {
+		return nil, nil
+	}
+	// The Broker delivery spec is translated to a pubsub retry policy in the
+	// manner defined in the following post:
+	// https://github.com/google/knative-gcp/issues/1392#issuecomment-655617873
+	p, _ := period.Parse(*spec.BackoffDelay)
+	minimumBackoff, _ := p.Duration()
+	var maximumBackoff time.Duration
+	switch *spec.BackoffPolicy {
+	case eventingduckv1beta1.BackoffPolicyLinear:
+		maximumBackoff = minimumBackoff
+	case eventingduckv1beta1.BackoffPolicyExponential:
+		maximumBackoff = defaultMaximumBackoff
+	default:
+		return nil, fmt.Errorf("Unrecognized backoff policy: %v", *spec.BackoffPolicy)
+	}
+	return &pubsub.RetryPolicy{
+		MinimumBackoff: minimumBackoff,
+		MaximumBackoff: maximumBackoff,
+	}, nil
+}
+
+// getDeadLetterTopic extracts the dead letter topic ID from its destination
+// URI. A dead letter topic should be specified as `pubsub://<dead-letter-topic-id>`.
+func getDeadLetterTopicID(sink *duckv1.Destination) (string, error) {
+	if sink.URI == nil {
+		return "", fmt.Errorf("Dead letter sink URI not specified")
+	}
+	if scheme := sink.URI.Scheme; scheme != "pubsub" {
+		return "", fmt.Errorf("Dead letter sink URI scheme should be pubsub, instead is %v", scheme)
+	}
+	topicID := sink.URI.Host
+	if topicID == "" {
+		return "", fmt.Errorf("Dead letter topic must not be empty")
+	}
+	if len(topicID) > 255 {
+		return "", fmt.Errorf("Dead letter topic maximum length is 255 characters: %v", topicID)
```

Comment:
These validations should live in pkg/apis/broker/v1beta1/broker_validation.go. That way they'll be part of the webhook and give the user immediate feedback.

Generally, if the object has invalid field values that will always be invalid, we should reject that in the webhook instead of letting it be created (and never recovering).